internal/service/batch/compute_environment.go

```diff
@@ -365,6 +344,170 @@ func resourceComputeEnvironmentUpdate(d *schema.ResourceData, meta interface{})
+					computeResourceUpdate.InstanceTypes = flex.ExpandStringSet(d.Get("compute_resources.0.instance_type").(*schema.Set))
+				}
+
+				if d.HasChange("compute_resources.0.launch_template.#") {
+					launchTemplate := &batch.LaunchTemplateSpecification{}
+					if launchTemplateCount := d.Get("compute_resources.0.launch_template.#"); launchTemplateCount.(int) > 0 {
+						// we previously didn't have a launch template and now we do
+						if launchTemplateId, ok := d.GetOk("compute_resources.0.launch_template.0.launch_template_id"); ok {
+							launchTemplate.LaunchTemplateId = aws.String(launchTemplateId.(string))
+						}
+						if launchTemplateName, ok := d.GetOk("compute_resources.0.launch_template.0.launch_template_name"); ok {
+							launchTemplate.LaunchTemplateName = aws.String(launchTemplateName.(string))
+						}
+						if launchTemplateVersion, ok := d.GetOk("compute_resources.0.launch_template.0.version"); ok {
+							launchTemplate.Version = aws.String(launchTemplateVersion.(string))
+						}
+					} else {
+						// To remove the custom launch template and use the default launch template,
+						// set launchTemplateId or launchTemplateName member of the launch template specification to an empty string.
+						launchTemplate.LaunchTemplateId = aws.String("")
+					}
+					computeResourceUpdate.LaunchTemplate = launchTemplate
+				} else if launchTemplateCount := d.Get("compute_resources.0.launch_template.#"); launchTemplateCount.(int) > 0 {
+					// we have a launch template configuration. check for changes to it
+					hasLaunchTemplateChange := false
+					launchTemplate := &batch.LaunchTemplateSpecification{}
+					if d.HasChange("compute_resources.0.launch_template.0.launch_template_id") {
+						hasLaunchTemplateChange = true
+						if launchTemplateId, ok := d.GetOk("compute_resources.0.launch_template.0.launch_template_id"); ok {
+							launchTemplate.LaunchTemplateId = aws.String(launchTemplateId.(string))
+						} else {
+							launchTemplate.LaunchTemplateId = aws.String("")
+						}
+					}
+					if d.HasChange("compute_resources.0.launch_template.0.launch_template_name") {
+						hasLaunchTemplateChange = true
+						if launchTemplateName, ok := d.GetOk("compute_resources.0.launch_template.0.launch_template_name"); ok {
+							launchTemplate.LaunchTemplateName = aws.String(launchTemplateName.(string))
+						} else {
+							launchTemplate.LaunchTemplateName = aws.String("")
+						}
+					}
+					if d.HasChange("compute_resources.0.launch_template.0.version") {
+						hasLaunchTemplateChange = true
+						if launchTemplateVersion, ok := d.GetOk("compute_resources.0.launch_template.0.version"); ok {
+							launchTemplate.Version = aws.String(launchTemplateVersion.(string))
+						} else {
+							launchTemplate.Version = aws.String("")
+						}
+					}
+					if hasLaunchTemplateChange {
+						computeResourceUpdate.LaunchTemplate = launchTemplate
+					}
+				}
+
+				if d.HasChange("compute_resources.0.tags") {
```

Comment:
Note for reviewers... I would really have liked to incorporate `tags_all` here but I was afraid of getting it wrong, so I decided to leave it out of scope for this PR. But with this PR it should become much more feasible in the future